### ISSUE TITLE
feat(website): Edit page should include a drop down for fields with defined options

### ISSUE THREAD

**Issue by anna-parker** [loculus-project/loculus#2347]
The values.yaml has fields with defined options, e.g. a list of valid countries (https://github.com/loculus-project/loculus/pull/2335 ): the website should also use this to display a dropdown on the submit/edit/revise pages.

~Give users a list of valid metadata options to choose from, or link to a list of valid fields in the metadata docs.~

~Potentially remove the country list on the webpage (for creating groups): /loculus/website/src/components/User/listOfCountries.ts and replace this with the countries in the yaml as well- we should be able to remove this and use the values from the yaml and add this check when editing sequences.~

**Comment by fhennig** [comment]
Yes I think following what Theo said, we can implement support for having a restricted set of options for a field. I think since it's then not just for preprocessing, I'd add a new `options` key to the metadata field type, and then pass that to the prepro config as well.

So:

```yaml
- name: ...
  displayName: ...
  options:
    - foo
    - bar
```

**Comment by chaoran-chen** [comment]
I think that we need to distinguish between the values that we would like users to provide as input (i.e., guideline-like, as it is with the `desired`) and the values that the pipeline accepts. E.g., the pipeline accepts both ["Czechia"](https://github.com/loculus-project/loculus/blob/feb3165f950cf1a811df2590a2f3a21ac0b536fe/kubernetes/loculus/values.yaml#L256) and ["Czech Republic"](https://github.com/loculus-project/loculus/blob/feb3165f950cf1a811df2590a2f3a21ac0b536fe/kubernetes/loculus/values.yaml#L477C17-L477C31) with the latter being the old name that users shouldn't provide anymore.

**Comment by theosanderson** [comment]
👍 happy either way - it would be (IMO) a big improvement over the current set up even without @chaoran-chen's nice refinement, but the refinement sounds nice

**Comment by fhennig** [comment]
Ah, good point! I think maybe we make the options objects too, and we add a 'deprecated' key:
```
- name: ...
  displayName: ...
  options:
    - name: Czechia
    - name: Czech Repulic
      deprecated: true
```

I think it'd also make sense to add the option of a `displayName`:


```
options:
  - name: nucleotideSeq
    displayName: Nucleotide Sequence
```

**Comment by anna-parker** [comment]
Regarding if we should mark countries as deprecated in the config I think perhaps that is out of scope - in the case of countries we should anyways supply an additional config from geonames with all the geoLocAdmin1 and geoLocAdmin2 regions of each countries, their hierarchies (i.e. relationships) and also dates of validity. I would suggest keeping the options list simple for the moment and only expanding later on.

A further reason for this is an uploader might upload older data which should be marked as Czech Republic still - so without also adding a date check might not give the correct options.

**Comment by fhennig** [comment]
> in the case of countries we should anyways supply an additional config from geonames with all the geoLocAdmin1 and geoLocAdmin2 regions of each countries, their hierarchies (i.e. relationships) and also dates of validity. I would suggest keeping the options list simple for the moment and only expanding later on.

I think I didn't understand the first half of this, but happy to keep it simple 😅 

It might also be fine to modify this again in the future, or accept different types of settings for the `options` key.

**Comment by chaoran-chen** [comment]
Yeah, the `displayName` solution sounds good. Then, it is possible to write something like "Czech Repulic (old name)" if we want. This is more flexible than having a deprecated flag.

**Comment by fhennig** [comment]
also if we already use objects instead of plain strings it's easy to add more keys to the object later in a non-breaking way
